project/apps/api-gateway/src/auth/auth.controller.ts

```diff
@@ -52,23 +59,31 @@ export class AuthController {
       httpOnly: true,
       secure: process.env.NODE_ENV === 'production',
       sameSite: 'lax',
-      maxAge: 60 * 60 * 24 * 7 * 1000,
+      maxAge: 60 * 60 * 1000, // 1 hour
+    });
+    
+    res.cookie('refresh_token', session.refresh_token, {
+      httpOnly: true,
+      secure: process.env.NODE_ENV === 'production',
+      sameSite: 'lax',
+      maxAge: 60 * 60 * 24 * 7 * 1000, // 1 week
     });
 
     return res.status(HttpStatus.OK).json({ userData });
   }
 
   @Post('signout')
   async signOut(@Res() res: Response) {
-    res.clearCookie('token');
+    res.clearCookie('access_token');
+    res.clearCookie('refresh_token');
     return res
       .status(HttpStatus.OK)
       .json({ message: 'Signed out successfully' });
   }
 
   @Get('me')
   async me(@Req() request: Request, @Res() res: Response) {
-    const token = request.cookies['token'];
+    const token = request.cookies['access_token'];
```

Comment:
for this endpoint do we need to let users refresh the token here? or what would happen if the token is expired. i think the current flow is we just throw an error here and log users out on the FE

actually on second thought, maybe we can use the `AuthGuard` here which will do it for us

Reply:
yep auth guard will do